gateway/middleware.go

```diff
@@ -0,0 +1,48 @@
+package gateway
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+func (tenants *Tenant) Authenticate(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logger := log.NewLogfmtLogger(os.Stdout)
```

Comment:
The logger can be shared with multiple routines, initialize it only once for Tenants and reuse it every time